pkg/gadgets/internal/socketenricher/bpf/sockets-map.h

```diff
@@ -67,7 +67,11 @@ struct {
 	__uint(max_entries, MAX_SOCKETS);
 	__type(key, struct sockets_key);
 	__type(value, struct sockets_value);
+#ifdef SOCKETS_MAP_IMPLEMENTATION
 } sockets SEC(".maps");
+#else
+} sockets SEC(".maps.auto");
+#endif
```

Comment:
Note: I am changing the macro `SOCKETS_MAP_IMPLEMENTATION` to something else in https://github.com/inspektor-gadget/inspektor-gadget/pull/1499 in order to support the socket enricher in tracing ebpf programs.

I am wondering about removing this `.auto` convention from the ELF section name and use a different convention directly with the map name, like a `ig_` prefix ( so the map would be named `ig_sockets`). Wdyt?

Also, after the "top ebpf" gadget, we tried to name ebpf programs and maps with the ig prefix, see https://github.com/inspektor-gadget/inspektor-gadget/pull/884.

Reply:
The naming with a `ig_` prefix sounds good. So `ig_sockets` or `ig_se_sockets` (`se` = socketenricher if only `ig_sockets` is too generic)